### ISSUE TITLE
[UX] add 'files/' as a field prefix for "File directory" location for image/file fields

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#2059]
When configuring a file or image field on an entity, one is able to specify into which directory on the filesystem files uploaded via this field can be placed. The description text for this field is as follows: 

```
Optional subdirectory within the upload destination where files will be stored. Do not include preceding or trailing slashes.
```

Not only is this text slightly obtuse and hard to follow, but the same explanation could be solved by adding a `#field_prefix` that pulls the Public file system path from settings, and shows the user exactly where their files will be placed.

**Comment by quicksketch** [comment]
Thanks @Graham-72! This PR looks good for public and private file storage, but contrib modules can add additional locations, such as an S3 bucket. I think it would be adequate to *only* support public files. Private files shouldn't have their absolute directory revealed anyway, and other storage locations might not be on the server at all.

**Comment by Dinornis** [comment]
I agree that it is adequate to only support public files for reasons stated.

**Comment by Graham-72** [comment]
I have updated the PR so that it only adds the prefix for public files https://github.com/backdrop/backdrop/pull/1680/commits/a0de9ac915cd33ec4e19a2c53ba97a1a68c8e4af.

I am now thinking that it would be helpful to include here on the form an indication whether the public or private folder is being used. These two screenshots show what is now displayed for public and then private locations:
![testimage backdrop starter 1](https://cloud.githubusercontent.com/assets/10559274/21586109/ac0b0642-d0c4-11e6-8ff3-736e86b0f9bc.png)
![testimage backdrop starter](https://cloud.githubusercontent.com/assets/10559274/21586111/b80d6890-d0c4-11e6-9eb8-09855ff15863.png)

**Comment by klonos** [comment]
> Private files shouldn't have their absolute directory revealed anyway...

If I get this correctly, this is to add prefixes under the "File directory" section in the file field setting form (`/admin/structure/types/manage/%content_type%/fields/%file_field%/edit`). Right? This page is not accessed by users with low level of permissions (like content authors for example), it is accessed by the person(s) building the website. They already have elevated permissions and are able to see and edit what the private directory is anyways, so why not show that information to them?

**Comment by quicksketch** [comment]
> They already have elevated permissions and are able to see and edit what the private directory is anyways, so why not show that information to them?

In most cases this is probably true, but not all. It's possible that a "Site builder" role would be able to add/configure fields but not be able to configure the file system, which is under a different permission for `administer site configuration`.

**Comment by quicksketch** [comment]
As this is a modest improvement even supporting public files (which already is about a 90% use-case), and the PR is simple, I've merged https://github.com/backdrop/backdrop/pull/1680. Thanks @Graham-72 for this! If we can come up with a way to support other stream wrappers, let's do it, but I don't think we have the APIs needed to get something like a "public display path" (and it might not be possible for the reason that not all stream wrappers are local in the first place).

**Comment by serundeputy** [comment]
This has been merged, but perhaps forgot to close?

I'm closing now ... feel free to re-open if necessary.
